core/cas-server-core-util-api/src/main/java/org/apereo/cas/util/RegexUtils.java

```diff
@@ -100,7 +100,12 @@ public static Pattern createPattern(final String pattern, final int flags) {
      * @see Matcher#matches()
      */
     public static boolean matches(final Pattern pattern, final String string) {
-        return pattern.matcher(string).matches();
+        try {
+            return pattern.matcher(string).matches();
+        } catch (NullPointerException npe) {
```

Comment:
Generally never a good idea to catch this. You should be simplify by using something like:

```
pattern != null and string != null and etc
```